tests/References/ResolvedReferenceTest.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\RST\References;
+
+use Doctrine\RST\References\ResolvedReference;
+use PHPUnit\Framework\TestCase;
+use function key;
+use function sprintf;
+
+class ResolvedReferenceTest extends TestCase
+{
+    /**
+     * @dataProvider attributesValid
+     *
+     * @param string[] $attributes
+     */
+    public function testCreateResolvedReference(array $attributes) : void
+    {
+        $resolvedReference = new ResolvedReference('title', 'url', ['title' => 'title'], $attributes);
+
+        $this->assertSame('title', $resolvedReference->getTitle());
+        $this->assertSame('url', $resolvedReference->getUrl());
+        $this->assertSame(['title' => 'title'], $resolvedReference->getTitles());
+        $this->assertSame($attributes, $resolvedReference->getAttributes());
+    }
+
+    public function attributesValid() : \Generator
+    {
+        yield 'attributes #1' => [
```

Comment:
Note: the `yield` approach is only useful when deferred data provider entries are required: favour a simple `array` when not needed.